### ISSUE TITLE
New design

### ISSUE THREAD

**Issue by pwalsh** [frictionlessdata/datapackage-ui#228]
# Description

- The old version of this app is at http://datapackagist.openknowledge.io . Most of the UI elements there are still relevant, as long as you think in terms of the "frame" and the "view" components.
- Please ref. #227 for additional context.

# Tasks

- [ ] Frame
  - [ ] Header / Footer
  - [ ] Friendly help
- [ ] Embeddable/reusable view component
  - [ ] Form (this is tricky, as it is a dynamically generated form, creating bootstrap form components from declarative code). The important thing, I think, is that form elements can be nested 1 - 3 levels. We need a design pattern around this (e.g.: just nest, or, "break out" at a certain depth into a new form. Breaking out will make lots of sense in some cases, like editing a [Table Schema](https://github.com/frictionlessdata/tableschema-ui), where we'd provide the Table Schema UI,a dn validating resources, as noted below)
  - [ ] Preview (a JSON previewer)
  - [ ] Actions (download, validate, validate resources)
  - [ ] Note that validate resource then embeds the goodtables report etc. simply not sure if we ignore this for the time being

**Comment by smth** [comment]
@pwalsh A couple of queries

- Is this tool going to continue to be known as Data Packagist?
- Could you provide some examples, or give instructions of how to generate examples, of how the form changes, and what it would mean to nest and break out the forms?

**Comment by pwalsh** [comment]
> Is this tool going to continue to be known as Data Packagist?

I don't think so. We might go for something more generic like "Data Package Builder" or something, but I do not know. Happy for input.

> Could you provide some examples, or give instructions of how to generate examples, of how the form changes, and what it would mean to nest and break out the forms?

- Nested: The `resources` array has nested resource objects. They are auto-generated in Data Packagist currently as item 1, item 2, etc.

<img width="1280" alt="screenshot 2017-07-25 12 28 52" src="https://user-images.githubusercontent.com/791678/28565437-f0c34d60-7134-11e7-93e9-899e6dd9561e.png">

- Break out: As nesting can get unwieldy fast, some aspects of the form can break out as modals with the "nestable" form object, rather than actually nesting them.

In the below screenshot, we have a resource item (see above), with its own nested table schema form (item 1 inside item 1). This is hard to follow visually, and hard to design for. At this level of nesting (3+), we could instead have a design pattern where we click "add a Table Schema", and it pops a modal with a dedicated for just for the schema.

<img width="1280" alt="screenshot 2017-07-25 12 30 53" src="https://user-images.githubusercontent.com/791678/28565532-3678b2be-7135-11e7-842c-c9e406efcc0a.png">

**Comment by smth** [comment]
In terms of branding, I was thinking something that will compliment goodtables, as a fellow Frictionless Data product.

![logos](https://user-images.githubusercontent.com/1375708/28582410-2902596e-715d-11e7-8c3b-25b68b14ea95.png)

Continuing that line of thinking, we can use a similar layout for this app. We can also borrow the "data cards" pattern that I created for OpenSpending, to give us something like this (the preview panel is collapsed, on the right).

![wireframe_01](https://user-images.githubusercontent.com/1375708/28582471-6a76240c-715d-11e7-9106-6b645a41e95c.png)

This would be realised in a colour scheme complimentary to [goodtables](http://p.smth.uk/goodtables/report/):

<img width="600" alt="colour" src="https://user-images.githubusercontent.com/1375708/28582815-ba50a442-715e-11e7-9710-eeb3a83e024f.png">

What do you think?

**Comment by pwalsh** [comment]
All looks good, but I am not sure the card model works here.

In OpenSpending, we have a data model many times more complex: basically, each "field" has multiple subfields, and this is consistent with every field that needs to be modelled.

Here, most fields are a simple plain `input` or `textarea` field, and some fields, such as `resources.resource` and `resources.resource.schema` are complex nested fields with subfields.

**Comment by smth** [comment]
I've realised that I missed some fields from each "item" (URL, Path, Data, Format, Media Type, Encoding), which is a challenge I need to consider, but I'm not sure if that's the problem you are highlighting here. The cards represent the schema, which still seems to me potentially a good fit, i.e consistent fields for each item, each item represents a table column.

Coming back to the "URL, Path, Data, Format, Media Type, Encoding" section, I'm confused by the "Data" field. All the others seem to be meta data about the resource, but "Data" sounds like a resource in itself.

**Comment by amercader** [comment]
My comments:

* Big +1 to align branding with goodtables.io

* It is purely a personal preference but I never liked the "packagist" name. What about "Data Packager"?It tells you exactly what it is and what you are going to create (a Data Package). 

* `Data` in the resource context was meant to be used to add the actual data in JSON format for that resource (eg `[{"a": 1, "b": 2}, {"a": 3, "b": 4}]`). TBH I would completely ignore it for this new version as it is a corner case.

* I like the grid for the schema fields (I guess that instead of "Add a description" it should say "Add a field"?) but my only concern is that there are previous steps before getting to the schema definition:
   - Upload an existing datapackage.json, or
   - Choose a profile
   - Enter dataset metadata
   - Add resource / Upload file
   - Add resource metadata
   - Describe schema
  What would be shown on the schema section meanwhile? Perhaps this should be expanded once you upload a file?

* Just a general thought but related to the above, what is the workflow that you expect from users? The one I described and the one from the current datapackagist is centered around first entering metadata about the dataset and then focusing on the file (same way that CKAN does it). But perhaps we should be focusing more on what users have and know about, which is a CSV, Excel, whatever (or more than one) and start with the files resources and then move to the dataset?

**Comment by smth** [comment]
Thanks @amercader 

The "Add" card should have been to either add an item or a special "description" card, but now that I realise there is more meta data, that might change.

I think, given the package meta data and resources are side by side, the user could start with either. With resources given prominence (and because it's probably the thing the user best understands), I would anticipate most people starting there. Before there are any resources added, there would be a prompt to start adding. Something like:

![wireframe_01_start_01](https://user-images.githubusercontent.com/1375708/28616629-54fd364a-71f5-11e7-949d-84e1194d60a0.png)

**Comment by smth** [comment]
First wireframe amended to include meta data:
![wireframe_01_resources-added_01](https://user-images.githubusercontent.com/1375708/28618821-c01b46e4-71fd-11e7-970f-de9d831827e5.png)

**Comment by amercader** [comment]
@smth I find that a bit confusing. Wouldn't the metadata fields make more sense as text boxes under Resource name?

**Comment by smth** [comment]
I was trying to avoid clutter. I guess though, as most of those fields are auto-generated, they could be hidden by default
![wireframe_01_resources-added_02](https://user-images.githubusercontent.com/1375708/28621549-4095c960-7209-11e7-9fdd-21b8b440d9ba.png)

and revealed with a click
![wireframe_01_resources-added_02_meta_01](https://user-images.githubusercontent.com/1375708/28621574-588e929a-7209-11e7-800a-607f41165d11.png)

**Comment by smth** [comment]
Moving naming conversation to [Slack](https://openknowledge.slack.com/archives/C0BCQ9ZT8/p1501074034216884)

**Comment by smth** [comment]
<img width="1366" alt="datapackagist_01" src="https://user-images.githubusercontent.com/1375708/28786754-1143a45e-7612-11e7-9b41-ebe9d510f537.png">

**Comment by smth** [comment]
Coded version of above: http://p.smth.uk/packagist/
Source: https://github.com/smth/packagist/tree/master
Build: https://github.com/smth/packagist/tree/gh-pages

**Comment by roll** [comment]
@smth 
@pwalsh 
Could we close it as done?
I've added links to Sam's work to https://github.com/frictionlessdata/datapackage-ui/issues/227
